quinn-udp/src/unix.rs

```diff
@@ -325,12 +357,63 @@ fn send(
     }
 }
 
-#[cfg(any(
-    target_os = "macos",
-    target_os = "ios",
-    target_os = "openbsd",
-    target_os = "netbsd"
-))]
+#[cfg(any(target_os = "macos", target_os = "ios",))]
+fn send(state: &UdpSocketState, io: SockRef<'_>, transmit: &Transmit<'_>) -> io::Result<()> {
+    let mut hdrs = unsafe { mem::zeroed::<[msghdr_x; BATCH_SIZE]>() };
+    let mut iovs = unsafe { mem::zeroed::<[libc::iovec; BATCH_SIZE]>() };
+    let mut ctrls = [cmsg::Aligned([0u8; CMSG_LEN]); BATCH_SIZE];
+    let addr = socket2::SockAddr::from(transmit.destination);
+    let segment_size = transmit.segment_size.unwrap_or(transmit.contents.len());
+    let cnt = transmit
+        .contents
+        .chunks(segment_size)
+        .enumerate()
+        .map(|(i, chunk)| {
+            prepare_msg(
+                &Transmit {
+                    destination: transmit.destination,
+                    ecn: transmit.ecn,
+                    contents: chunk,
+                    segment_size: Some(chunk.len()),
+                    src_ip: transmit.src_ip,
+                },
+                &addr,
+                &mut hdrs[i],
+                &mut iovs[i],
+                &mut ctrls[i],
+                true,
+                state.sendmsg_einval(),
+            );
```

Comment:
Nit: It is not idiomatic to execute a side-effect in an `Iterator::map`.

> If you’re doing some sort of looping for a side effect, it’s considered more idiomatic to use [for](https://doc.rust-lang.org/stable/book/ch03-05-control-flow.html#looping-through-a-collection-with-for) than map().

https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.map

While this might seem dogmatic, I do find reasoning about an Iterator chain much easier, when being able to rely on the convention of no-side-effects-in-map.

Instead I suggest either using a standard `for` loop here, or `Iterator::for_each`, updating `cnt` on each iteration.

Reply:
Can do either. What do the maintainers prefer?

Reply:
Yes, agreed that a `for` loop would be better here.